profiles/fscloud/variables.tf

```diff
@@ -0,0 +1,206 @@
+  type        = string
+  default     = "Manager"
+}
+
+variable "cos_instance_name" {
+  description = "The name to give the cloud object storage instance that will be provisioned by this module. Only required if 'create_cos_instance' is true."
+  type        = string
+  default     = null
+}
+
+variable "cos_location" {
+  description = "Location to provision the cloud object storage instance. Only used if 'create_cos_instance' is true."
+  type        = string
+  default     = "global"
+}
+
+variable "cos_tags" {
+  description = "Optional list of tags to be added to cloud object storage instance. Only used if 'create_cos_instance' it true."
+  type        = list(string)
+  default     = []
+}
+
+variable "existing_cos_instance_id" {
+  description = "The ID of an existing cloud object storage instance. Required if 'var.create_cos_instance' is false."
+  type        = string
+  default     = null
+}
+
+variable "cos_plan" {
+  description = "Plan to be used for creating cloud object storage instance. Only used if 'create_cos_instance' it true."
+  type        = string
+  default     = "standard"
+  validation {
+    condition     = contains(["standard", "lite"], var.cos_plan)
+    error_message = "The specified cos_plan is not a valid selection!"
+  }
+}
+
+##############################################################################
+# COS bucket variables
+##############################################################################
+
+variable "create_cos_bucket" {
+  description = "Set as true to create a new Cloud Object Storage bucket"
+  type        = bool
+  default     = true
+}
+
+variable "primary_region" {
+  description = "region for the primary bucket"
+  type        = string
+  default     = "us-south"
+}
+
+variable "secondary_region" {
+  description = "region for the secondary bucket"
+  type        = string
+  default     = "us-east"
+}
+
+variable "bucket_name" {
+  type        = string
+  description = "The name to give the newly provisioned COS bucket. Only required if 'create_cos_bucket' is true."
+  default     = null
+}
+
+variable "bucket_storage_class" {
+  type        = string
+  description = "the storage class of the newly provisioned COS bucket. Only required if 'create_cos_bucket' is true. Supported values are 'standard', 'vault', 'cold', and 'smart'."
+  default     = "standard"
+
+  validation {
+    condition     = can(regex("^standard$|^vault$|^cold$|^smart$", var.bucket_storage_class))
+    error_message = "Variable 'bucket_storage_class' must be 'standard', 'vault', 'cold', or 'smart'."
+  }
+}
+
+variable "activity_tracker_crn" {
+  type        = string
+  description = "Activity tracker crn for COS bucket (Optional)"
+  default     = null
+}
+
+variable "sysdig_crn" {
+  type        = string
+  description = "Sysdig Monitoring crn for COS bucket (Optional)"
```

Comment:
It shouldn't be optional if 'create_cos_bucket' is true though right? So perhaps update the variable description here

Reply:
Infact, we probably need logic to ensure Sysdig is passed in when creating a bucket, because I think currently null could be passed in even when 'create_cos_bucket' is true